### ISSUE TITLE
Provide an option to increase/attach shared memory for specific ContainerOp

### ISSUE THREAD

**Issue by arunnalpet** [kubeflow/pipelines#6880]
### What steps did you take

While creating a ContainerOp, there is no option to attach a volume for /dev/shm. This is needed for training pytorch models. Training with larger batch size, throws up this error. Training with tiny batch sizes will go through just fine. 

### What happened:
Currently the code fails with: 
`Bus error. It is possible that dataloader's workers are out of shared memory. Please try to raise your shared memory limit.
`
### What did you expect to happen:
Either shared memory should be mounted automatically for container operations, or we should have an option to configure shared memory using pipeline DSL. I do not see any examples as well around this.

### Anything else you would like to add:
This is taken care while spinning up notebook instances with "Enabled Shared Memory" Option. The same option needs to be provided with ContainerOp in pipeline DSL. The same is discussed in this [issue thread](https://github.com/kubeflow/kubeflow/issues/2522#issuecomment-520190909).

### Labels

/area backend
/area sdk

/area components

---

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by chensun** [comment]
Not sure if there's anything special with `/dev/shm`. But KFP DSL v1 supports `add_volume` and `add_volume_mount`. Would that work for you?

**Comment by b02202050** [comment]
Hi @arunnalpet, I encountered this problem too, and I resolved as follows.

```python
import kfp.dsl as dsl
import kubernetes as k8s # should pip install kubernetes first

@dsl.pipeline(name="test_training")
def training_pipeline():
    volume = dsl.PipelineVolume(volume=k8s.client.V1Volume(
        name="shm",
        empty_dir=k8s.client.V1EmptyDirVolumeSource(medium='Memory')))

    training_op.add_pvolumes({'/dev/shm': volume})

```

Hope this helps.

*Rerference: https://stackoverflow.com/a/63806311*
*As for why medium is set to **Memory** is described [here](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir)*

**Comment by arunnalpet** [comment]
Thanks @b02202050 .. let me try this out, and will close the issue subsequently..

**Comment by kbegiedza** [comment]
Adding @b02202050 code results in `This step is in Error state with this message: OOMKilled (exit code 137)` after training step in `wait` container.
